src/preamble.js

```diff
@@ -233,22 +232,17 @@ function allocate(slab, types, allocator, ptr) {
 
   var singleType = typeof types === 'string' ? types : null;
 
-  var ret;
-  if (allocator == ALLOC_NONE) {
-    ret = ptr;
-  } else {
-    ret = [{{{ ('_malloc' in IMPLEMENTED_FUNCTIONS) ? '_malloc' : 'null' }}},
+  var ret = [{{{ ('_malloc' in IMPLEMENTED_FUNCTIONS) ? '_malloc' : 'null' }}},
 #if DECLARE_ASM_MODULE_EXPORTS
-    stackAlloc,
+  stackAlloc,
```

Comment:
why deindent here? i think it should be indented higher than the `var ret` which it is nested under.

overall could be
```javascript
var ret = [
  ..malloc..
  ..stack.. // note indentation
][allocator](..);
```

Reply:
That means it was wrong previously then too?  Since it was aligned with ret before...

Anyway I'm hoping to remove all this code as part of this set of changes.

Reply:
but the diff shows you removed whitespace before `stackAlloc`? and the ret didn't move up or down?

Reply:
(doesn't matter much anyhow though)

Reply:
The ret did move out one scope, but it also got var added to the beginning.

If you look at the old code it was basically this:

```
ret = [
xxx
yyy
]
```

Note the lack indentation of xxx, yyy.. in the old code.

Reply:
Oh ok, looks like it was wrong before then...